Feat: Added Konnect examples for Auth plugins related to Consumers

d19dotca opened the pull request:
### Description

A customer of Kong Konnect discovered that we were missing data in our documentation on the API calls in Konnect to make for adding credentials to consumers.

Added example API calls to the documentation for plugins including HMAC, JWT, Basic Auth, Key Auth, and ACL.

I believe these will eventually be added elsewhere as part of the ongoing work in the API spec documents for Konnect, but as a stop-gap I am hoping it can exist in the individual plugin documentation for now.

Related: [Slack thread](https://kongstrong.slack.com/archives/C03NRECFJPM/p1704731546097239)

### Testing instructions

Preview link: https://deploy-preview-6742--kongdocs.netlify.app/

### Checklist 

- [x] Review label added <!-- (see below) -->
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

Guaris commented on app/_hub/kong-inc/acl/how-to/_index.md:
````diff
@@ -54,6 +54,26 @@ acls:
+associate a group to a consumer using the following request:
+
+```bash
+curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER}/acls \
````
```suggestion
curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER_ID}/acls \
```

Guaris commented on app/_hub/kong-inc/acl/how-to/_index.md:
````diff
@@ -54,6 +54,26 @@ acls:
+    --data "tags[]=tag2"
+```
+
+`CONSUMER`: The `username` property of the consumer entity to associate the credentials with.
````
```suggestion
`CONSUMER_ID`: The UUID of the consumer entity to associate the credentials with.
```

Guaris commented on app/_hub/kong-inc/hmac-auth/overview/_index.md:
````diff
@@ -81,6 +88,16 @@ hmacauth_credentials:
 ```
 {% endnavtab %}
+{% navtab Konnect %}
+You can provision new username/password credentials by making the following
````
```suggestion
You can provision new username and password credentials by making the following
```

Guaris commented on app/_hub/kong-inc/jwt/overview/_index.md:
````diff
@@ -93,6 +100,30 @@ jwt_secrets:
+You can provision a new HS256 JWT credential by issuing the following HTTP request:
+
+```bash
+curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER}/jwt -H "Content-Type: application/x-www-form-urlencoded" \
````
```suggestion
curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER_ID}/jwt -H "Content-Type: application/x-www-form-urlencoded" \
```

Guaris commented on app/_hub/kong-inc/acl/how-to/_index.md:
```diff
@@ -54,6 +54,26 @@ acls:
 * `tags`: Optional descriptor tags for the group.
+{% endnavtab %}
+{% navtab Konnect %}
+After you have added an authentication plugin to a Service or a Route, and you have
+created your [consumers](/gateway/latest/admin-api/#consumer-object), you can now
+associate a group to a consumer using the following request:
```
```suggestion
After you have added an authentication plugin to a service or a route, and you have
created the [consumers object](/gateway/latest/admin-api/#consumer-object), you can now
associate a group to a consumer using the following request:
```

Guaris commented on app/_hub/kong-inc/basic-auth/overview/_index.md:
````diff
@@ -69,6 +78,17 @@ basicauth_credentials:
+You can provision new username/password credentials by making the following HTTP request:
+
+```bash
+curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER}/basic-auth \
````
```suggestion
curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER_ID}/basic-auth \
```

Guaris commented on app/_hub/kong-inc/hmac-auth/overview/_index.md:
````diff
@@ -81,6 +88,16 @@ hmacauth_credentials:
+HTTP request:
+
+```bash
+curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER}/hmac-auth \
````
```suggestion
curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER_ID}/hmac-auth \
```

Guaris commented on app/_hub/kong-inc/key-auth/overview/_index.md:
````diff
@@ -120,6 +128,15 @@ keyauth_credentials:
+Provision new credentials by making the following HTTP request:
+
+```bash
+curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER}/key-auth
````
```suggestion
curl -X POST https://{us|eu}.api.konghq.com/v2/control-planes/{controlPlaneId}/core-entities/consumers/{CONSUMER_ID}/key-auth
```

Guaris commented:
@d19dotca Some changes to the URL's to explicitly state they require the UUID of the consumer in the Konnect API.

d19dotca commented:
Thanks @Guaris - I committed the suggested changes now. 👍

Guaris commented:
I'll test this tomorrow morning and merge it!

lena-larionova reviewed (commented):
@Guaris @d19dotca 
I think @mheap may have published a similar change already: https://github.com/Kong/docs.konghq.com/pull/6758

Does that satisfy the request, or is this PR still neccessary?

cloudjumpercat commented:
@lena-larionova @Guaris After looking at the content added in this PR and what was added in #6758, it looks like they cover most of the same information (Michael's PR doesn't include ACL though, it only includes auth plugins, and this PR does). However, because in the original plugin docs, sometimes consumers are talked about in both overview and the basic config examples pages, this PR updates the overview pages and Michael's updates the basic config pages. I think before we merge this PR we should:
* Remove any duplicate consumer content from the overview and maybe instead link to the basic config example page in it's place so the information is only in one place. 
* Make sure we reuse Michael's basic config examples consumer content in the ACL plugin page.

Guaris commented:
@d19dotca @cloudjumpercat Dustin, based on Diana's feedback would these changes still solve the initial customer concerns?

d19dotca commented:
Sorry for the late reply @Guaris, @cloudjumpercat, and @lena-larionova. 

I updated the data based on where they were at the time, but perhaps this is also a good opportunity to follow what I think Michael was trying to which is move these all from the Overview page to their own config example page for managing consumers and credentials for the plugin. 

It sounds like mine may contain more data (at least since ACL was also added in my PR), so perhaps it makes sense to just validate that Michael and I came to the same examples (and correct if any mismatches) but also move them to their own page in the Config Examples area at the same time as it sounds like Michael was doing? I'm not sure the best way forward since it sounds like we unfortunately duplicated efforts on that, not sure which path is the best path forward in this case.

d19dotca commented:
Actually if we're going to move to the Config Examples pages from Overview, then maybe it's easier to use Michael's PR and we can just update to also include ACL?

d19dotca commented:
Okay yeah I think this can be closed, and then Michael's can be approved and merged, and I'll add in afterwards the ACL one to that.
